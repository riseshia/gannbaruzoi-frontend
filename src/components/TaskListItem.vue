<template>
  <li class="task">
    <input type="checkbox" v-model="instanceStatus">

    <label>
      ( {{ loggedSize }} /<span contenteditable @keydown.enter.prevent.stop="updateEstimatedSize">
      {{ estimatedSize }}
      </span>)
      <span contenteditable @keydown.enter.prevent.stop="updateDescription">
        {{ description }}</span>
      <a href="#" @click.prevent="deleteTask">[x]</a>
    </label>
  </li>
</template>

<script>
export default {
  name: 'task-list-item',

  props: {
    taskId: Number,
    description: String,
    estimatedSize: Number,
    type: String,
    loggedSize: Number,
    status: Boolean
  },
  data () {
    return {
      instanceStatus: this.status
    }
  },
  watch: {
    instanceStatus: {
      handler: 'onInstanceStatusChanged',
      immediate: true,
      deep: false
    }
  },
  methods: {
    onInstanceStatusChanged (newVal) {
      this.$emit('updateTask', { id: this.taskId, status: newVal })
    },
    updateDescription (e) {
      this.$emit('updateTask', {
        description: e.target.innerText,
        id: this.taskId
      })
    },
    updateEstimatedSize (e) {
      this.$emit('updateTask', {
        stimatedSize: e.target.innerText,
        id: this.taskId
      })
    },
    deleteTask () {
      this.$emit('deleteTask', { id: this.taskId })
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
li {
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
