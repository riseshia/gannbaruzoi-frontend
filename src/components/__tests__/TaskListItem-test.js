import { shallow } from 'vue-test-utils'
import TaskListItem from '@/components/TaskListItem'

describe('TaskListItem.vue', () => {
  const propsData = {
    taskId: 1,
    description: '今日も一日かんばるぞい',
    estimatedSize: 5,
    loggedSize: 0,
    status: false
  }
  const wrapper = shallow(TaskListItem, { propsData })
  it('renders correct contents', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
