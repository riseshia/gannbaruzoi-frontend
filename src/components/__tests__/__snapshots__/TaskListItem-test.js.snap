// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem.vue renders correct contents 1`] = `
"<li class=\\"task\\"><input type=\\"checkbox\\"> <label>
    ( 0 /<span contenteditable=\\"true\\">
    5
    </span>)
    <span contenteditable=\\"true\\">
      今日も一日かんばるぞい</span> <a href=\\"#\\">[x]</a></label></li>"
`;
